app/controllers/welcome_controller.rb

```diff
@@ -190,32 +196,27 @@ def create_radio_jockey(row, row_index, xlsx, column_mapping, header_mapping)
       un_apr: row[column_mapping[header_mapping[:un_apr]]].to_s || "",
       un_may: row[column_mapping[header_mapping[:un_may]]].to_s || ""
     }
+  end
 
-    existing_rj = RadioJockey.find_by(show_name: show_name)
-    if existing_rj
-      puts "Found an existing RJ with Show Name : ", show_name
-      update = false
-
-      if member_type != existing_rj.member_type
-        if member_type == "Returning RJ" # Returning has higher priority, else ignore
-          update = true
-        end
-      else # Same member type
-        if (semesters_in_kanm > existing_rj.semesters_in_kanm) ||
-          (semesters_in_kanm == existing_rj.semesters_in_kanm && expected_grad < existing_rj.expected_grad) ||
-          (semesters_in_kanm == existing_rj.semesters_in_kanm && expected_grad == existing_rj.expected_grad && timestamp < existing_rj.timestamp)
-          update = true
-        end
-      end
-
+  def should_update_rj(existing_rj, row, column_mapping, header_mapping)
+    member_type = row[column_mapping[header_mapping[:member_type]]].to_s || ""
```

Comment:
Aren't you doing this in "prepare_rj_data" function?

Reply:
Yes, but I didn't want to pass the entire rj_data structure to this method or these 4 values as parameters, as that might also trigger a code climate issue. Hence, I went with this approach. If it causes any code climate duplications, I shall refactor again.

Reply:
Addressed comments by removing duplication